tests/unit/test_consumer.py

```diff
@@ -358,3 +359,89 @@ def test_rules_files_with_invalid_yaml_are_dropped_and_produce_an_error(self):
         logger_output = "\n".join(logger.output)
         self.assertIn("tab.rule", logger_output)
         self.assertIn("multicolon.rule", logger_output)
+
+    def test_rules_have_correct_labels(self):
+        unlabeled_rule = {
+            "groups": [
+                {
+                    "name": "alert_on_error",
+                    "rules": [
+                        {
+                            "alert": "alert_on_error",
+                            "expr": 'rate({%%juju_topology%%}|="ERROR"[5m]) > 0',
+                            "for": "1m",
+                            "labels": {
+                                "severity": "critical",
+                            },
+                            "annotations": {"summary": "Logs found at ERROR level"},
+                        }
+                    ],
+                }
+            ]
+        }
+        self.sandbox.writetext("error.rules", yaml.dump(unlabeled_rule))
+        self.harness.begin_with_initial_hooks()
+        relation = self.harness.charm.model.get_relation("logging")
+        rules = json.loads(relation.data[self.harness.charm.app].get("alert_rules"))
+        import pprint
+
+        pprint.pprint(rules)
+        expr = rules["groups"][0]["rules"][0]["expr"]
+        self.assertTrue(
+            "juju_model" in expr
+            and "juju_model_uuid" in expr
+            and "juju_application" in expr
+            and "juju_charm" in expr
+            and "juju_unit" not in expr
+        )
+        self.assertEqual(
```

Comment:
We can use inequality to reduce fragility:
```suggestion
        self.assertGreaterEqual(
```

Reply:
I don't understand this suggestion. Could you elaborate?

Reply:
If the lib at some point in the future adds another label that is independent to this feature, then this test would break because it would no longer match `{"juju_application", "juju_charm", "juju_model", "juju_model_uuid", "severity"}` exactly. For that reason I am proposing to change the assertion to "subset" (⊆)

Reply:
I think it will be much better to update the test when it changes. These bugs make me think it is best to check for exact labels.